xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -259,33 +262,31 @@ bool CProcessorHD::InitProcessor()
     }
   }
 
-  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
-
-  if (SUCCEEDED(m_pEnumerator.As(&pEnumerator1)))
+  if (SUCCEEDED(m_pEnumerator.As(&m_pEnumerator1)))
```

Comment:
I'm not sure about this change because processor is Re-Init when is lost and is used `pEnumerator` to track if is valid: 

https://github.com/xbmc/xbmc/blob/a3e7d1a9b0f3da68f1571eb3981e74920f536a2f/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp#L383-L403

If we use an old instance of  `m_pEnumerator1` maybe is invalid reference or cause deadlock, etc. Think is more safe at it is now.

Reply:
I made sure it's constructed / destructed in sync with the existing m_pEnumerator that it is based on. It's protected by the same locks, so seems to me we won't have situation where one is valid and not the other one (under Windows 10 with API is available).
Still, it's possible to move the creation a bit earlier so there are no return statements between m_pEnumerator  creation and m_pEnumerator1 creation.
Also possible to take the lock when entering IsFormatConversionSupported and ListSupportedConversions, called from out of class.